File: sherlock_project/pass_scan.rb
def password_strength(password)
  # Initialize score
  score = 0

  # Check length
  score += 1 if password.length >= 8

  # Check for uppercase letters
  score += 1 if password =~ /[A-Z]/

  # Check for lowercase letters
  score += 1 if password =~ /[a-z]/

  # Check for digits
  score += 1 if password =~ /\d/

  # Check for special characters
  score += 1 if password =~ /[\W_]/

  score
end

def evaluate_password(password)
  score = password_strength(password)

  case score
  when 5
    "Very Strong"
  when 4
    "Strong"
  when 3
    "Moderate"
  when 2
    "Weak"
  else
    "Very Weak"
  end
end

def main
  puts "Enter a password to evaluate its strength:"
  password = gets.chomp

  strength = evaluate_password(password)
  puts "Password strength: #{strength}"
end

main

File: sherlock_project/pass_scan.java
import java.util.Scanner;

/**
* Calculates the strength of a password based on various criteria including length, uppercase letters, lowercase letters, digits, and special characters.
* 
* @param password The password to evaluate its strength.
* @return An integer representing the strength score of the password.
*/
public class PasswordScanner {

    public static int passwordStrength(String password) {
        int score = 0;

        // Check length
        if (password.length() >= 8) {
            score += 1;
        }

        // Check for uppercase letters
        if (password.matches(".*[A-Z].*")) {
            score += 1;
        }

        // Check for lowercase letters
        if (password.matches(".*[a-z].*")) {
            score += 1;
        }

        // Check for digits
        if (password.matches(".*\\d.*")) {
            score += 1;
        }

        // Check for special characters
        if (password.matches(".*[\\W_].*")) {
            score += 1;
        }

        return score;
    }

    public static String evaluatePassword(String password) {
        int score = passwordStrength(password);

        switch (score) {
            case 5:
                return "Very Strong";
            case 4:
                return "Strong";
            case 3:
                return "Moderate";
            case 2:
                return "Weak";
            default:
                return "Very Weak";
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a password to evaluate its strength:");
        String password = scanner.nextLine();

        String strength = evaluatePassword(password);
        System.out.println("Password strength: " + strength);
    }
}